tasks/post_copy_request_to_queue/post_copy_request_to_queue.py

```diff
@@ -0,0 +1,218 @@
+        logging.critical(message)
+        raise Exception(message)
+
+
+# Define our exponential delay function
+# maybe move to shared library or somewhere else?
+def exponential_delay(base_delay: int, exponential_backoff: int = 2) -> int:
+    """
+    Exponential delay function. This function is used for retries during failure.
+    Args:
+        base_delay: Number of seconds to wait between recovery failure retries.
+        exponential_backoff: The multiplier by which the retry interval increases during each attempt.
+    Returns:
+        An integer which is multiplication of base_delay and exponential_backoff.
+    Raises:
+        None
+    """
+    delay = base_delay + (random.randint(0, 1000) / 1000.0)
+    time.sleep(delay)
+    LOGGER.debug(f"Performing back off retry sleeping {delay} seconds")
+    return base_delay * exponential_backoff
+
+
+def handler(event: Dict[str, Any], context: None) -> None:
+    """
+    Lambda handler. This lambda calls the task function to perform db queries
+    and send message to SQS.
+
+        Environment Vars:
+            PREFIX (string): the prefix
+            DATABASE_PORT (string): the database port. The standard is 5432.
+            DATABASE_NAME (string): the name of the database.
+            DATABASE_USER (string): the name of the application user.
+            DB_QUEUE_URL (string): the SQS URL for status-update-queue
+            RECOVERY_QUEUE_URL (string): the SQS URL for staged_recovery_queue
+        Parameter store:
+            {prefix}-drdb-host (string): host name that will be retrieved from secrets manager
+            {prefix}-drdb-user-pass (string):db password that will be retrieved from secrets manager
+    Args:
+        event:
+            A dictionary from the S3 bucket. See schemas/input.json for more information.
+        context: An object required by AWS Lambda. Unused.
+    Returns:
+        None
+    Raises:
+        Exception: If unable to retrieve the SQS URLs or exponential retry fields from env variables.
+
+    """
+    LOGGER.setMetadata(event, context)
+
+    # retrieving values from the env variables
+    backoff_env = [
+        "DB_QUEUE_URL",
+        "RECOVERY_QUEUE_URL",
+        "MAX_RETRIES",
+        "RETRY_SLEEP_SECS",
+        "RETRY_BACKOFF",
+    ]
+    backoff_args = []
+    for var in backoff_env:
+        env_var_value = os.getenv(var, None)
+        if env_var_value is None or len(env_var_value) == 0:
+            message = f"{var} is not set and is required"
+            LOGGER.critical(message)
+            raise Exception(message)
+
+        if var in ["MAX_RETRIES", "RETRY_SLEEP_SECS", "RETRY_BACKOFF"]:
+            try:
+                env_var_value = int(env_var_value)
+            except ValueError as ve:
+                LOGGER.critical(f"{var} must be set to an integer.")
+                raise ve
+        backoff_args.append(env_var_value)
+
+    records = event["Records"][0]
```

Comment:
the event is actually coming from s3 rather than SQS.
for every object restored in s3 from glacier, it will send one event to the lambda instead of a list. 

I discussed with @AndrewDorn  regarding this. Check his [comment](https://github.com/nasa/cumulus-orca/pull/93#discussion_r638296999)